verification/spec/connection-handshake/Environment.tla

```diff
@@ -0,0 +1,233 @@
+
+
+VARIABLES
+    bufChainA,      \* A buffer (sequence) for messages inbound to chain A.
+    bufChainB,      \* A buffer for messages inbound to chain B.
+    storeChainA,    \* The local store of chain A.
+    storeChainB,    \* The local store of chain B.
+    maliciousEnv    \* If TRUE, environment interferes w/ CH protocol. 
+
+
+chainAParameters == [
+    connectionID |-> { "connAtoB" },
+    clientID |-> { "clientChainA" }
+]
+
+chainBParameters == [
+    connectionID |-> { "connBtoA" },
+    clientID |-> { "clientChainB" }
+]
+
+
+ClientIDs == { chainAParameters.clientID, chainBParameters.clientID }
+ConnectionIDs == { chainAParameters.connectionID, chainBParameters.connectionID }
+
+chainAVars == <<bufChainA, bufChainB, storeChainA>>
+chainBVars == <<bufChainA, bufChainB, storeChainB>>
+chainStoreVars == <<storeChainA, storeChainB>>
+allVars == <<chainStoreVars, bufChainA, bufChainB, maliciousEnv>>
+
+
+chmA == INSTANCE ConnectionHandshakeModule
+        WITH MaxHeight      <- MaxHeight,
+             inBuf          <- bufChainA,
+             outBuf         <- bufChainB,
+             store          <- storeChainA,
+             ConnectionIDs  <- chainAParameters.connectionID,
+             ClientIDs      <- chainAParameters.clientID
+
+
+chmB == INSTANCE ConnectionHandshakeModule
+        WITH MaxHeight      <- MaxHeight,
+             inBuf          <- bufChainB,      \* Flip the message buffers w.r.t. chain A buffers.
+             outBuf         <- bufChainA,      \* Inbound for "A" is outbound for "B".
+             store          <- storeChainB,
+             ConnectionIDs  <- chainBParameters.connectionID,
+             ClientIDs      <- chainBParameters.clientID
+
+
+ConnectionStates == {"UNINIT", "INIT", "TRYOPEN", "OPEN"}
+
+
+ConnectionParameters ==
+    [
+        localEnd : chmA!ConnectionEnds,
+        remoteEnd : chmB!ConnectionEnds
+    ]
+
+Connections ==
+    [
+        parameters : ConnectionParameters,
+        state : ConnectionStates
+    ]
+
+Proofs ==
+    [
+        height : 1..MaxHeight
+    ]
+
+(******************************** Messages ********************************
+ These messages are connection handshake specific.
+ 
+ In the low-level connection handshake protocol, the four messages have the
+ following types: ConnOpenInit, ConnOpenTry, ConnOpenAck, ConnOpenConfirm.
+ These are described in ICS 003.
+ In this high-level specification, we choose slightly different names, to
+ make an explicit distinction to the low-level protocol. Message types
+ are as follows: CHMsgInit, CHMsgTry, CHMsgAck, and CHMsgConfirm. Notice that
+ the fields of each message are also different to the ICS 003 specification.
+ ***************************************************************************)
+ConnectionHandshakeMessages ==
+    [type : {"CHMsgInit"}, 
+     parameters : ConnectionParameters]
+    \union
+    [type : {"CHMsgTry"},
+     parameters : ConnectionParameters,
+     stateProof : Proofs,
```

Comment:
This proof is about connection. In my view it is better naming it like that, i.e., connectionProof for example.